robot/robot_test.go

```diff
@@ -281,6 +281,99 @@ func Test_RobotCannotMoveThroughObstacles(t *testing.T) {
 	r.PrintEnvironment()
 }
 
+func Test_RobotCannotMoveThroughObstaclesInItsWayNorth(t *testing.T) {
+	r := Robot{}
+	r.EnvMap.SquareSize = 1
+	r.EnvMap.Map = `
+###########
+#         #
+#      #  #
+#         #
+###########`
+	r.Position = Position{X: 7, Y: 3, Orientation: North}
+
+	expectedMap := `
+###########
+#         #
+#      #  #
+#      ▲  #
+###########`
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+
+	r.MoveModule = &Mock{t: t, expected: Centimeters(2)}
+	err := r.Move(2)
+
+	if err == nil {
+		t.Errorf("Expected error message")
+	}
+
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+}
+
+func Test_RobotCannotMoveThroughObstaclesInItsWayBack(t *testing.T) {
+	r := Robot{}
+	r.EnvMap.SquareSize = 1
+	r.EnvMap.Map = `
+###########
+#         #
+#      #  #
+#         #
+###########`
+	r.Position = Position{X: 7, Y: 1, Orientation: North}
+
+	expectedMap := `
+###########
+#      ▲  #
+#      #  #
+#         #
+###########`
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+
+	r.MoveModule = &Mock{t: t, expected: Centimeters(2)}
+	err := r.Move(-2)
+
+	if err == nil {
+		t.Errorf("Expected error message")
+	}
+
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+}
+
+func Test_RobotCannotMoveThroughObstaclesInItsWayEast(t *testing.T) {
+	r := Robot{}
+	r.EnvMap.SquareSize = 1
+	r.EnvMap.Map = `
+###########
+#         #
+#      #  #
+#         #
+###########`
+	r.Position = Position{X: 6, Y: 2, Orientation: East}
+
+	expectedMap := `
+###########
+#         #
+#     ▶#  #
+#         #
+###########`
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+
+	r.MoveModule = &Mock{t: t, expected: Centimeters(2)}
+	err := r.Move(2)
+
+	if err == nil {
+		t.Errorf("Expected error message")
+	}
+
+	r.PrintModule = &Mock{t: t, expected: expectedMap}
+	r.PrintEnvironment()
+}
+
 func Test_RobotWithoutEnvironmentalMapCanNotMoveAtAll(t *testing.T) {
```

Comment:
Please add another test Test_RobotCannotMoveThroughObstaclesInItsWayWest.
An it would be great to a have only one other test where the obstacle is more than one step away from the robots current position.